manifests/config.pp

```diff
@@ -0,0 +1,82 @@
+# == Class oath::config
+#
+# This class is called from oath for pam_oath configuration.
+# This class ensures files and directories have the correct 
+# selinux contexts to run (rpm defaults for pam_oath are nonexistant)
+# It also parses the oath::oath_users hieradata and calls oath::config::user.pp
+# define to check types and create a concat fragment that will be inserted into
+# the /etc/liboath/users.oath file.
+
+
+class oath::config {
+  assert_private()
+
+  # Ensures correct selinux context of config directory
+  file { '/etc/liboath':
+    ensure  => 'directory',
+    owner   => 'root',
+    group   => 'root',
+    seluser => 'system_u',
+    seltype => 'var_auth_t',
+  }
+
+  file { '/etc/liboath/exclude_users.oath':
+    ensure  => 'file',
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    seluser => 'system_u',
+    seltype => 'var_auth_t',
+    source  => "puppet:///modules/${module_name}/etc/liboath/exclude_users.oath",
```

Comment:
We need to template out the `exclude_users` and `exclude_groups` files and expose the parameters via `init.pp`

Reply:
I'd be happy to make this change. I will push an update in the next few days.

Reply:
Currently the `users.oath` file is built with concat instead of a template. Should the `exclude_users` and `exclude_groups` files be built with concat, a template or does it not matter?

Reply:
@zach-simp Yeah, `exclude_users` and `exclude_groups` should be `concat` built.